能提供一个关闭 AVX2 的镜像，好痛苦

## Enhancement
你们迫不及待就把这个 issue 关闭：https://github.com/StarRocks/starrocks/issues/60543

我按照编译的方式尝试了有一周了，仍然不能成功，白白浪费这么多时间。请体会下使用者的心情。

docker run --name main --hostname starrocks-host -v /etc/localtime:/etc/localtime:ro -v /Users/mac/workspace/opensrc/.m2:/root/.m2 -v /Users/mac/workspace/opensrc/starrocks:/root/starrocks -it -d starrocks/dev-env-ubuntu:latest

cd /root/starrocks ,关闭 AVX2 进行 ./build.sh


[  9%] Building CXX object src/formats/CMakeFiles/Formats.dir/parquet/page_reader.cpp.o
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h: In lambda function:
/root/starrocks/be/src/simd/delta_decode.h:63:43: warning: AVX vector return without AVX enabled changes the ABI [-Wpsabi]
   63 |         v8i x = _mm256_loadu_si256((v8i*)p);
      |                                           ^
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:933:1: error: inlining failed in call to ‘always_inline’ ‘void _mm256_storeu_si256(__m256i_u*, __m256i)’: target specific option mismatch
  933 | _mm256_storeu_si256 (__m256i_u *__P, __m256i __A)
      | ^~~~~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:67:28: note: called from here
   67 |         _mm256_storeu_si256((v8i*)p, x);
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:47,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avx2intrin.h:119:1: error: inlining failed in call to ‘always_inline’ ‘__m256i _mm256_add_epi32(__m256i, __m256i)’: target specific option mismatch
  119 | _mm256_add_epi32 (__m256i __A, __m256i __B)
      | ^~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:66:29: note: called from here
   66 |         x = _mm256_add_epi32(x, _mm256_slli_si256(x, 8));
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:47,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avx2intrin.h:655:1: error: inlining failed in call to ‘always_inline’ ‘__m256i _mm256_slli_si256(__m256i, int)’: target specific option mismatch
  655 | _mm256_slli_si256 (__m256i __A, const int __N)
      | ^~~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:66:29: note: called from here
   66 |         x = _mm256_add_epi32(x, _mm256_slli_si256(x, 8));
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:47,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avx2intrin.h:119:1: error: inlining failed in call to ‘always_inline’ ‘__m256i _mm256_add_epi32(__m256i, __m256i)’: target specific option mismatch
  119 | _mm256_add_epi32 (__m256i __A, __m256i __B)
      | ^~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:65:29: note: called from here
   65 |         x = _mm256_add_epi32(x, _mm256_slli_si256(x, 4));
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:47,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avx2intrin.h:655:1: error: inlining failed in call to ‘always_inline’ ‘__m256i _mm256_slli_si256(__m256i, int)’: target specific option mismatch
  655 | _mm256_slli_si256 (__m256i __A, const int __N)
      | ^~~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:65:29: note: called from here
   65 |         x = _mm256_add_epi32(x, _mm256_slli_si256(x, 4));
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:47,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avx2intrin.h:119:1: error: inlining failed in call to ‘always_inline’ ‘__m256i _mm256_add_epi32(__m256i, __m256i)’: target specific option mismatch
  119 | _mm256_add_epi32 (__m256i __A, __m256i __B)
      | ^~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:64:29: note: called from here
   64 |         x = _mm256_add_epi32(x, v_min_delta);
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:43,
                 from /root/starrocks/be/src/gutil/strings/fastmem.h:21,
                 from /root/starrocks/be/src/util/memcmp.h:41,
                 from /root/starrocks/be/src/util/int96.h:21,
                 from /root/starrocks/be/src/util/hash_util.hpp:54,
                 from /root/starrocks/be/src/runtime/decimalv2_value.h:47,
                 from /root/starrocks/be/src/column/column_visitor.h:19,
                 from /root/starrocks/be/src/column/column.h:22,
                 from /root/starrocks/be/src/formats/parquet/encoding_bss.h:19,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/11/include/avxintrin.h:927:1: error: inlining failed in call to ‘always_inline’ ‘__m256i _mm256_loadu_si256(const __m256i_u*)’: target specific option mismatch
  927 | _mm256_loadu_si256 (__m256i_u const *__P)
      | ^~~~~~~~~~~~~~~~~~
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h:63:35: note: called from here
   63 |         v8i x = _mm256_loadu_si256((v8i*)p);
      |                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:56:1: note: in expansion of macro ‘MFV_AVX2’
   56 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
make[2]: *** [src/formats/CMakeFiles/Formats.dir/build.make:818: src/formats/CMakeFiles/Formats.dir/parquet/encoding.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  9%] Generating orc_proto.pb.h, orc_proto.pb.cc
[  9%] Generating bit_packing_gen.inc


这个我是 vim build.sh 把 USE_AVX2 = OFF 改成 off，
之前是直接 ./build.sh --without-AVX2 ，然后 ./run-be-ut.sh
也是报错：

[ 68%] Linking CXX executable storage/task/engine_storage_migration_task_test
[ 99%] Built target starrocks_test_objs
[ 99%] Building CXX object test/CMakeFiles/compaction_task_test.dir/storage/lake/compaction_task_test.cpp.o
collect2: fatal error: ld terminated with signal 9 [Killed]
compilation terminated.
make[2]: *** [test/CMakeFiles/engine_storage_migration_task_test.dir/build.make:335: test/storage/task/engine_storage_migration_task_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:1500: test/CMakeFiles/engine_storage_migration_task_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 99%] Linking CXX executable storage/lake/compaction_task_test
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm3sys15PrintStackTraceERNS_11raw_ostreamEi+0x31)[0x7fd29ed12d01]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(_ZN4llvm3sys17RunSignalHandlersEv+0xee)[0x7fd29ed10a3e]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xe40236)[0x7fd29ed13236]
/lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x7fd29d9b9520]
/bin/ld.lld(_ZN3lld3elf12InputSection16relocateNonAllocIN4llvm6object7ELFTypeILNS3_7support10endiannessE1ELb1EEENS4_12Elf_Rel_ImplIS8_Lb1EEEEEvPhNS3_8ArrayRefIT0_EE+0x1ae)[0x5bac9e]
/bin/ld.lld(_ZN3lld3elf12InputSection7writeToIN4llvm6object7ELFTypeILNS3_7support10endiannessE1ELb1EEEEEvPh+0xbc)[0x5af5fc]
/bin/ld.lld[0x5e6c43]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xdb88bc)[0x7fd29ec8b8bc]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xdb8785)[0x7fd29ec8b785]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xdb7c4a)[0x7fd29ec8ac4a]
/lib/x86_64-linux-gnu/libLLVM-14.so.1(+0xdb791c)[0x7fd29ec8a91c]
/lib/x86_64-linux-gnu/libstdc++.so.6(+0xdc253)[0x7fd29dd83253]
/lib/x86_64-linux-gnu/libc.so.6(+0x94ac3)[0x7fd29da0bac3]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x44)[0x7fd29da9ca04]
collect2: fatal error: ld terminated with signal 7 [Bus error]
compilation terminated.
make[2]: *** [test/CMakeFiles/compaction_task_test.dir/build.make:335: test/storage/lake/compaction_task_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:1553: test/CMakeFiles/compaction_task_test.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

这些本来速度就都很慢，非常痛苦，能不能帮忙提供一个关闭 AVX2 的镜像，方便启动 be 的

> [@jg-8](https://github.com/jg-8) it seems like an issue with lambdas. lambdas do not inherit target-specific attributes from their enclosing functions. may be u can modify the code as below and try.
> 
> <img alt="Image" width="1026" height="901" src="https://private-user-images.githubusercontent.com/22505528/465161449-2dc17781-ceb2-4871-81ea-edec909cf0eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIyMjQ5OTQsIm5iZiI6MTc1MjIyNDY5NCwicGF0aCI6Ii8yMjUwNTUyOC80NjUxNjE0NDktMmRjMTc3ODEtY2ViMi00ODcxLTgxZWEtZWRlYzkwOWNmMGViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzExVDA5MDQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwODc3NDUyMzE4NjJhMDE0YjZkZTEwMjEzMDQ0MGMxODcwZjc4YWIyZDEyYTAxMjRhZTg5NTlhZmM2MTU0MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Im2qV514tGuQPl7qqII5KsMHvMoJk8zA0W1OUBxQf3E">

改完运行到 9% 时能过，不过到 14% 时又报错

[ 14%] Building CXX object src/formats/CMakeFiles/Formats.dir/parquet/encoding.cpp.o
In file included from /root/starrocks/be/src/simd/delta_decode.h:17,
                 from /root/starrocks/be/src/formats/parquet/encoding_delta.h:25,
                 from /root/starrocks/be/src/formats/parquet/encoding.cpp:24:
/root/starrocks/be/src/simd/delta_decode.h: In function ‘void starrocks::delta_prefix_avx2(int32_t*, int32_t)’:
/root/starrocks/be/src/simd/delta_decode.h:57:24: error: ‘x，v_min_delta’ was not declared in this scope; did you mean ‘v_min_delta’?
   57 |     x=_mm256_add_epi32(x，v_min_delta);
      |                        ^~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:53:1: note: in expansion of macro ‘MFV_AVX2’
   53 | MFV_AVX2(void delta_prefix_avx2(int32_t* p,int32_t min_delta) {
      | ^~~~~~~~
/root/starrocks/be/src/simd/delta_decode.h:58:25: error: ‘x，_mm256_slli_si256’ was not declared in this scope; did you mean ‘_mm256_slli_si256’?
   58 |     x= _mm256_add_epi32(x，_mm256_slli_si256(x,4));
      |                         ^~~~~~~~~~~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:53:1: note: in expansion of macro ‘MFV_AVX2’
   53 | MFV_AVX2(void delta_prefix_avx2(int32_t* p,int32_t min_delta) {
      | ^~~~~~~~
/root/starrocks/be/src/simd/delta_decode.h:55:15: warning: unused variable ‘v_min_delta’ [-Wunused-variable]
   55 |     const v8i v_min_delta = _mm256_set1_epi32(min_delta);
      |               ^~~~~~~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:53:1: note: in expansion of macro ‘MFV_AVX2’
   53 | MFV_AVX2(void delta_prefix_avx2(int32_t* p,int32_t min_delta) {
      | ^~~~~~~~
/root/starrocks/be/src/simd/delta_decode.h: In function ‘void starrocks::delta_decode_chain_int32_avx2(int32_t*, int, int32_t, int32_t&)’:
/root/starrocks/be/src/simd/delta_decode.h:92:13: error: ‘prefix’ was not declared in this scope; did you mean ‘profil’?
   92 |             prefix(buf + i);
      |             ^~~~~~
/root/starrocks/be/src/simd/multi_version.h:25:32: note: in definition of macro ‘MFV_IMPL’
   25 |             ATTR static inline IMPL _Pragma("GCC diagnostic pop")
      |                                ^~~~
/root/starrocks/be/src/simd/delta_decode.h:66:1: note: in expansion of macro ‘MFV_AVX2’
   66 | MFV_AVX2(void delta_decode_chain_int32_avx2(int32_t* buf, int n, int32_t min_delta, int32_t& last_value) {
      | ^~~~~~~~
make[2]: *** [src/formats/CMakeFiles/Formats.dir/build.make:818: src/formats/CMakeFiles/Formats.dir/parquet/encoding.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:747: src/formats/CMakeFiles/Formats.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 14%] Building CXX object src/exec/CMakeFiles/Exec.dir/exchange_node.cpp.o
[ 15%] Building CXX object src/exec/CMakeFiles/Exec.dir/scan_node.cpp.o
[ 15%] Building CXX object src/exec/CMakeFiles/Exec.dir/select_node.cpp.o


不知道能不能过

> [@jg-8](https://github.com/jg-8) at line 57,58,58 the ，after x is incorrect. looks like u have input chinese character or something like that. please use the below code. MFV_AVX2(void delta_prefix_avx2(int32_t* p, int32_t min_delta) { using v8i = __m256i; const v8i v_min_delta = _mm256_set1_epi32(min_delta); v8i x = _mm256_loadu_si256((v8i*)p); x = _mm256_add_epi32(x, v_min_delta); x = _mm256_add_epi32(x, _mm256_slli_si256(x, 4)); x = _mm256_add_epi32(x, _mm256_slli_si256(x, 8)); _mm256_storeu_si256((v8i*)p, x); });

@rohitrs1983 thanks very much, i has changed the code.

但是构建好多个小时，到现在才构建到 55%，并且又失败了

其中有：
[ 44%] Building CXX object src/exprs/CMakeFiles/Exprs.dir/agg/factory/aggregate_resolver_stream.cpp.o
/root/starrocks/be/src/gen_cpp/build/gen_cpp/opcode/builtin_functions.cpp: In function ‘void __static_initialization_and_destruction_0(int, int)’:
/root/starrocks/be/src/gen_cpp/build/gen_cpp/opcode/builtin_functions.cpp:968:1: note: variable tracking size limit exceeded with ‘-fvar-tracking-assignments’, retrying without
  968 | }
      | ^
[ 44%] Building CXX object src/gen_cpp/CMakeFiles/StarRocksGen.dir/build/gen_cpp/RuntimeFilter_types.cpp.o


并且在：
[ 55%] Building CXX object src/exprs/CMakeFiles/Exprs.dir/time_functions.cpp.o
/root/starrocks/be/src/exprs/time_functions.cpp:18:10: fatal error: libdivide.h: No such file or directory
   18 | #include <libdivide.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/exprs/CMakeFiles/Exprs.dir/build.make:1056: src/exprs/CMakeFiles/Exprs.dir/time_functions.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:851: src/exprs/CMakeFiles/Exprs.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/query_cache_action.cpp.o
[ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/datacache_action.cpp.o
[ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/pipeline_blocking_drivers_action.cpp.o
[ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/lake/dump_tablet_metadata_action.cpp.o
[ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/stop_be_action.cpp.o
[ 55%] Linking CXX static library ../../../output/tmp/RELEASE/libWebserver.a
[ 55%] Built target Webserver
make: *** [Makefile:156: all] Error 2

real	172m11.449s
user	214m25.184s
sys	22m29.419s

又再次失败了

> > > [@jg-8](https://github.com/jg-8) at line 57,58,58 the ，after x is incorrect. looks like u have input chinese character or something like that. please use the below code. MFV_AVX2(void delta_prefix_avx2(int32_t* p, int32_t min_delta) { using v8i = __m256i; const v8i v_min_delta = _mm256_set1_epi32(min_delta); v8i x = _mm256_loadu_si256((v8i*)p); x = _mm256_add_epi32(x, v_min_delta); x = _mm256_add_epi32(x, _mm256_slli_si256(x, 4)); x = _mm256_add_epi32(x, _mm256_slli_si256(x, 8)); _mm256_storeu_si256((v8i*)p, x); });
> > 
> > 
> > [@rohitrs1983](https://github.com/rohitrs1983) thanks very much, i has changed the code.
> > 但是构建好多个小时，到现在才构建到 55%，并且又失败了
> > 其中有： [ 44%] Building CXX object src/exprs/CMakeFiles/Exprs.dir/agg/factory/aggregate_resolver_stream.cpp.o /root/starrocks/be/src/gen_cpp/build/gen_cpp/opcode/builtin_functions.cpp: In function ‘void __static_initialization_and_destruction_0(int, int)’: /root/starrocks/be/src/gen_cpp/build/gen_cpp/opcode/builtin_functions.cpp:968:1: note: variable tracking size limit exceeded with ‘-fvar-tracking-assignments’, retrying without 968 | } | ^ [ 44%] Building CXX object src/gen_cpp/CMakeFiles/StarRocksGen.dir/build/gen_cpp/RuntimeFilter_types.cpp.o
> > 并且在： [ 55%] Building CXX object src/exprs/CMakeFiles/Exprs.dir/time_functions.cpp.o /root/starrocks/be/src/exprs/time_functions.cpp:18:10: fatal error: libdivide.h: No such file or directory 18 | #include <libdivide.h> | ^~~~~~~~~~~~~ compilation terminated. make[2]: *** [src/exprs/CMakeFiles/Exprs.dir/build.make:1056: src/exprs/CMakeFiles/Exprs.dir/time_functions.cpp.o] Error 1 make[1]: *** [CMakeFiles/Makefile2:851: src/exprs/CMakeFiles/Exprs.dir/all] Error 2 make[1]: *** Waiting for unfinished jobs.... [ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/query_cache_action.cpp.o [ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/datacache_action.cpp.o [ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/pipeline_blocking_drivers_action.cpp.o [ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/lake/dump_tablet_metadata_action.cpp.o [ 55%] Building CXX object src/http/CMakeFiles/Webserver.dir/action/stop_be_action.cpp.o [ 55%] Linking CXX static library ../../../output/tmp/RELEASE/libWebserver.a [ 55%] Built target Webserver make: *** [Makefile:156: all] Error 2
> > real 172m11.449s user 214m25.184s sys 22m29.419s
> > 又再次失败了
> 
> [@jg-8](https://github.com/jg-8) as part of [#60331](https://github.com/StarRocks/starrocks/pull/60331) new third party lib has been added recently seems like you have not updated ur third-party libs. you need to build the third-party lib libdivide.

[ 15%] Building CXX object src/exprs/CMakeFiles/Exprs.dir/time_functions.cpp.o
/root/starrocks/be/src/exprs/time_functions.cpp: In static member function ‘static starrocks::StatusOr<starrocks::Cow<starrocks::Column>::ImmutPtr<starrocks::Column> > starrocks::TimeFunctions::hour_from_unixtime(starrocks::FunctionContext*, const Columns&)’:
/root/starrocks/be/src/exprs/time_functions.cpp:1610:34: error: ‘class cctz::time_zone’ has no member named ‘lookup_offset’
 1610 |                 int offset = ctz.lookup_offset(t).offset;
      |                                  ^~~~~~~~~~~~~
make[2]: *** [src/exprs/CMakeFiles/Exprs.dir/build.make:1056: src/exprs/CMakeFiles/Exprs.dir/time_functions.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:851: src/exprs/CMakeFiles/Exprs.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

又出现新错了

擦，真是崩溃呀！！！
本来速度就极慢！！！
先 download-thirdparty.sh 下载完，
再
cd ${STARROCKS_HOME}/thirdparty/src/cctz-2.3;
patch -p1 < ${STARROCKS_HOME}/patches/cctz_02_lookup_offset.patch;
然后
build-thirdparty.sh

结果一天一夜后，快完成时，又报下来这个错误，还让不让活！！！


src/time_zone_info.cc:778:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
  778 | TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      | ^~~~~~~~~~~~
src/time_zone_info.cc:778:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:783:28: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
  783 | time_zone::absolute_lookup TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      |                            ^~~~~~~~~~~~
src/time_zone_info.cc:783:28: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:791:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) const’
  791 | TimeZoneInfo::LookupOffset(const time_point<seconds> &tp) const {
      | ^~~~~~~~~~~~
src/time_zone_info.cc:791:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) cons’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:836:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
  836 | TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      | ^~~~~~~~~~~~
src/time_zone_info.cc:836:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:841:28: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
  841 | time_zone::absolute_lookup TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      |                            ^~~~~~~~~~~~
src/time_zone_info.cc:841:28: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:849:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) const’
  849 | TimeZoneInfo::LookupOffset(const time_point<seconds> &tp) const {
      | ^~~~~~~~~~~~
src/time_zone_info.cc:849:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) cons’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
make: *** [<builtin>: time_zone_info.o] Error 1
make: *** Waiting for unfinished jobs....

===== begin build cctz-2.3
/bin/../bin/g++ -O3 -fno-omit-frame-pointer -Wno-class-memaccess -fPIC -g -gz=zlib -ffile-prefix-map=/root/starrocks/thirdparty/src=. -ffile-prefix-map=/root/starrocks/thirdparty/installed=. -g -Wall -Iinclude -std=c++11  -fPIC -MMD -I /root/starrocks/thirdparty/installed/include  -c -o time_zone_info.o src/time_zone_info.cc
/bin/../bin/g++ -O3 -fno-omit-frame-pointer -Wno-class-memaccess -fPIC -g -gz=zlib -ffile-prefix-map=/root/starrocks/thirdparty/src=. -ffile-prefix-map=/root/starrocks/thirdparty/installed=. -g -Wall -Iinclude -std=c++11  -fPIC -MMD -I /root/starrocks/thirdparty/installed/include  -c -o time_zone_lookup.o src/time_zone_lookup.cc
src/time_zone_info.cc:778:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
  778 | TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      | ^~~~~~~~~~~~
src/time_zone_info.cc:778:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:783:28: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
  783 | time_zone::absolute_lookup TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      |                            ^~~~~~~~~~~~
src/time_zone_info.cc:783:28: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:791:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) const’
  791 | TimeZoneInfo::LookupOffset(const time_point<seconds> &tp) const {
      | ^~~~~~~~~~~~
src/time_zone_info.cc:791:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) cons’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:836:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
  836 | TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      | ^~~~~~~~~~~~
src/time_zone_info.cc:836:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::TransitionType&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:841:28: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
  841 | time_zone::absolute_lookup TimeZoneInfo::Offset(std::int_fast64_t unix_time,
      |                            ^~~~~~~~~~~~
src/time_zone_info.cc:841:28: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::Offset(int_fast64_t, const cctz::Transition&) const’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
src/time_zone_info.cc:849:1: error: no declaration matches ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) const’
  849 | TimeZoneInfo::LookupOffset(const time_point<seconds> &tp) const {
      | ^~~~~~~~~~~~
src/time_zone_info.cc:849:1: note: no functions named ‘cctz::time_zone::absolute_lookup cctz::TimeZoneInfo::LookupOffset(cctz::time_point<std::chrono::duration<long int> >&) cons’
In file included from src/time_zone_info.cc:33:
src/time_zone_info.h:61:7: note: ‘class cctz::TimeZoneInfo’ defined here
   61 | class TimeZoneInfo : public TimeZoneIf {
      |       ^~~~~~~~~~~~
/bin/../bin/g++ -O3 -fno-omit-frame-pointer -Wno-class-memaccess -fPIC -g -gz=zlib -ffile-prefix-map=/root/starrocks/thirdparty/src=. -ffile-prefix-map=/root/starrocks/thirdparty/installed=. -g -Wall -Iinclude -std=c++11  -fPIC -MMD -I /root/starrocks/thirdparty/installed/include  -c -o time_zone_posix.o src/time_zone_posix.cc
make: *** [<builtin>: time_zone_info.o] Error 1

WBD，又是这个错误！！！！

--- a/be/src/simd/multi_version.h
+++ b/be/src/simd/multi_version.h
@@ -24,22 +24,58 @@
     _Pragma("GCC diagnostic push") _Pragma("GCC diagnostic ignored \"-Wunused-function\"") \
             ATTR static IMPL _Pragma("GCC diagnostic pop")
 
+#if defined(__SSE4_2__)
 #define MFV_SSE42(IMPL) MFV_IMPL(IMPL, __attribute__((target("sse4.2"))))
+#endif
+
+#if defined(__AVX2__)
 #define MFV_AVX2(IMPL) MFV_IMPL(IMPL, __attribute__((target("avx2"))))
+#endif
+
+#if defined(__AVX512F__)
 #define MFV_AVX512F(IMPL) MFV_IMPL(IMPL, __attribute__((target("avx512f"))))
+#endif
+
+#if defined(__AVX512BW__)
 #define MFV_AVX512BW(IMPL) MFV_IMPL(IMPL, __attribute__((target("avx512bw"))))
+#endif
+
+#if defined(__AVX512VL__)
 #define MFV_AVX512VL(IMPL) MFV_IMPL(IMPL, __attribute__((target("avx512vl"))))
+#endif
+
+#if defined(__AVX512F__) && defined(__AVX512BW__) && defined(__AVX512VL__)
 #define MFV_AVX512VLBW(IMPL) MFV_IMPL(IMPL, __attribute__((target("avx512f,avx512vl,avx512bw"))))
+#endif
+
 #define MFV_DEFAULT(IMPL) MFV_IMPL(IMPL, __attribute__((target("default"))))
 
-#else
+#endif // end of defined(__GNUC__) && defined(__x86_64__)
 
+#if !defined(MFV_SSE42)
 #define MFV_SSE42(IMPL)
-#define MFV_AVX2(IMPL)
-#define MFV_AVX512F(IMPL)
-#define MFV_AVX512BW(IMPL)
-#define MFV_AVX512VL(IMPL)
-#define MFV_AVX512VLBW(IMPL)
-#define MFV_DEFAULT(IMPL) IMPL
+#endif
 
+#if !defined(MFV_AVX2)
+#define MFV_AVX2(IMPL)
+#endif
+
+#if !defined(MFV_AVX512F)
+#define MFV_AVX512F(IMPL)
+#endif
+
+#if !defined(MFV_AVX512BW)
+#define MFV_AVX512BW(IMPL)
+#endif
+
+#if !defined(MFV_AVX512VL)
+#define MFV_AVX512VL(IMPL)
+#endif
+
+#if !defined(MFV_AVX512VLBW)
+#define MFV_AVX512VLBW(IMPL)
+#endif
+
+#if !defined(MFV_DEFAULT)
+#define MFV_DEFAULT(IMPL) IMPL
 #endif

